pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -221,10 +219,11 @@ protected void handleLookup(CommandLookupTopic lookup) {
         if (lookupSemaphore.tryAcquire()) {
             final String originalPrincipal = lookup.hasOriginalPrincipal() ? lookup.getOriginalPrincipal()
                     : this.originalPrincipal;
-            if (!validateOriginalPrincipal(originalPrincipal,
-                    newLookupErrorResponse(ServerError.AuthorizationError,
-                            "Valid Proxy Client role should be provided for lookup ", requestId),
-                    topicName.toString(), "Valid Proxy Client role should be provided for lookup ")) {
+            if (invalidOriginalPrincipal(originalPrincipal)) {
+                final String msg = "Valid Proxy Client role should be provided for lookup ";
```

Comment:
Instead of repeating the logging and send error, we could do like in https://github.com/apache/incubator-pulsar/blob/28ce0adaef76c52341231104daac4df3e2c55f53/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L1161